### DISCUSSION TITLE
Preferred way of transforming a `Kramdown::Document`

### DISCUSSION THREAD

**Discussion by suetanvil** [gettalong/kramdown#816] [Q&A] [answered]
I'm writing a literate programming thing that combines Markdown with embedded code.  I would like to parse the initial input file to a `Kramdown::Document`, make various changes to the tree, and then use a `Converter` to produce HTML.

It looks to me like the easiest way to do this is create the Document, use `root` to get the root node and then modify it in place, either by adding or removing child nodes where needed or by modifying the node values.

Is this the official supported way, or is there something better that I'm missing?

**Comment by gettalong** [comment, marked as answer]
Yes, this is the way to go. See for example the [RemoveHtmlTags converter](https://github.com/gettalong/kramdown/blob/master/lib/kramdown/converter/remove_html_tags.rb).

**Reply by suetanvil** [reply]
Thank you.

Also, thanks for your work with Kramdown. It's a really nice library.
